File: examples/python/check_bank_statement.py
"""Check that a bank statement balances before you import it.

myocr reads the statement and runs one arithmetic check:
opening balance + credits - debits must equal the closing balance printed on it.

    pip install myocr-client
    export MYOCR_API_KEY=sk_test_...
    python check_bank_statement.py ../sample_bank_statement.pdf
"""
import sys

from myocr_client import MyOCRClient

path = sys.argv[1] if len(sys.argv) > 1 else "../sample_bank_statement.pdf"
client = MyOCRClient()  # reads MYOCR_API_KEY

result = client.convert(path, model="bank_statement", output="json")
statement = result.data["bank_statement"]
check = result.data["reconciliation"]

print(f"{statement['bank_name']} - {statement['period']} ({statement['currency']})")
print(f"Transactions read: {check['n_transactions']}")
print(f"Opening {check['opening']:.2f} + credits {check['credits']:.2f} "
      f"- debits {check['debits']:.2f} = {check['computed_closing']:.2f}")
print(f"Closing balance on the statement: {check['closing']:.2f}")

if check["ok"]:
    print("CONSISTENT: safe to import.")
else:
    print(f"NOT CONSISTENT: off by {check['delta']:.2f}. Check the statement before importing.")
    sys.exit(1)

File: examples/node/check-bank-statement.mjs
// Check that a bank statement balances before you import it.
//
//   npm install myocr-client
//   export MYOCR_API_KEY=sk_test_...
//   node check-bank-statement.mjs ../sample_bank_statement.pdf
import { MyOCRClient } from 'myocr-client';

const path = process.argv[2] ?? '../sample_bank_statement.pdf';
const client = new MyOCRClient(); // reads MYOCR_API_KEY

const result = await client.convert(path, { model: 'bank_statement', output: 'json' });
const statement = result.data.bank_statement;
const check = result.data.reconciliation;

console.log(`${statement.bank_name} - ${statement.period} (${statement.currency})`);
console.log(`Transactions read: ${check.n_transactions}`);
console.log(`Opening ${check.opening.toFixed(2)} + credits ${check.credits.toFixed(2)} - debits ${check.debits.toFixed(2)} = ${check.computed_closing.toFixed(2)}`);
console.log(`Closing balance on the statement: ${check.closing.toFixed(2)}`);

if (check.ok) {
  console.log('CONSISTENT: safe to import.');
} else {
  console.log(`NOT CONSISTENT: off by ${check.delta.toFixed(2)}. Check the statement before importing.`);
  process.exitCode = 1;
}
